workflow/usage.rst

```diff
@@ -0,0 +1,159 @@
+                        - published
+                    transitions:
+                        to_review:
+                            from: draft
+                            to:   review
+                        publish:
+                            from: review
+                            to:   published
+                        reject:
+                            from: review
+                            to:   rejected
+
+
+.. code-block: php
+
+    class BlogPost
+    {
+        // This property is used by the marking store
+        public $currentPlace;
+        public $title;
+        public $content
+    }
+
+.. note::
+
+    The marking store type could be "property_accessor" or "scalar".
+    A scalar marking type does not support a model being on multiple places.
+
+With this workflow named ``blog_publishing`` you can get help to decide
+what actions that are allowed on a blog post. ::
+
+    $post = new \BlogPost();
+
+    $workflow = $this->container->get('workflow.blog_publishing');
+    $workflow->can($post, 'publish'); // False
+    $workflow->can($post, 'to_review'); // True
+
+    // Update the currentState on the post
+    try {
+        $workflow->apply($post, 'to_review');
+    } catch (LogicException $e) {
+        // ...
+    }
+
+    // See all the available transition for the post in the current state
+    $transitions = $workflow->getEnabledTransitions($post);
+
+Using Events
+------------
+
+To make your workflows even more powerful you could construct the ``Workflow``
+object with an ``EventDispatcher``. You can now create event listeners to
+block transitions ie depending on the data in the blog post. The following
+events are dispatched:
+
+* ``workflow.guard``
+* ``workflow.[workflow name].guard``
+* ``workflow.[workflow name].guard.[transition name]``
+
+See example to make sure no blog post without title is moved to "review"::
+
+    class BlogPostReviewListener implements EventSubscriberInterface
+    {
+        public function guardReview(GuardEvent $event)
+        {
+            /** @var Acme\BlogPost $post */
+            $post = $event->getSubject();
+            $title = $post->title;
+
+            if (empty($title)) {
+                // Posts with no title should not be allowed
+                $event->setBlocked(true);
+            }
+        }
+
+        public static function getSubscribedEvents()
+        {
+            return array(
+                'workflow.blogpost.guard.to_review' => array('guardReview'),
+            );
+        }
+    }
+
+With help from the ``EventDispatcher`` and the ``AuditTrailListener`` you
+could easily enable logging::
+
+    $logger = new PSR3Logger();
+    $subscriber = new AuditTrailListener($logger);
+    $dispatcher->addSubscriber($subscriber);
+
+Usage in Twig
+-------------
+
+Using your workflow in your Twig templates reduces the need of domain logic
+in the view layer. Consider this example of the control panel for our blog's
```

Comment:
"[...] for our blog's edit page" -> "[...] of the blog." ? (again, to remove first person)